### PR TITLE
Feature/us6318 hide onsite group participants

### PR DESCRIPTION
**Pull request by Doug-Shannon**
onsite groups shouldn't show participants anywhere, unless you are a leader.

### CODE REVIEW

File: Gateway/crds-angular/Controllers/API/GroupController.cs [hunk 1]
```diff
@@ -316,6 +316,37 @@ public IHttpActionResult GetGroups(int groupTypeId)
...
+        /// Return typeID of group
+        /// </summary>
+        /// <param name="groupId">An integer identifying the group that we want to check the type of.</param>
+        /// <returns>MyGroup</returns>
+        [RequiresAuthorization]
+        [ResponseType(typeof(int))]
+        [VersionedRoute(template: "group/{groupId}/getGroupType", minimumVersion: "1.0.0")]
+        [Route("group/{groupId}/getGroupType")]
+        [HttpGet]
+        public IHttpActionResult GetGroupType(int groupId)
```
Comment (Doug-Shannon) [inline Gateway/crds-angular/Controllers/API/GroupController.cs]: I'll switch it to yours

File: Gateway/crds-angular/Services/GroupToolService.cs [hunk 2]
```diff
@@ -684,6 +688,16 @@ public List<GroupDTO> SearchGroups(int[] groupTypeIds, string keywords = null, s
             return groups;
         }
 
+        public List<GroupDTO> GetGroupToolGroups(string token)
+        {
+            var groups = _groupService.GetGroupsForAuthenticatedUser(token, new int[] { _smallGroupTypeId, _onsiteGroupTypeId });
+            var participantRecord = _participantRepository.GetParticipantRecord(token);
+
+
```
Comment (Doug-Shannon) [inline Gateway/crds-angular/Services/GroupToolService.cs]: :|

File: crossroads.net/app/group_tool/services/groupDetail.service.js [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+
+    //From groupDetail.participants.controller.js loadGroupParticipants()
+    //participants array stored on this service
+
+    canAccessParticipants(isLeader, groupTypeID) {
+        let onsiteGroup = groupTypeID == CONSTANTS.GROUP.GROUP_TYPE_ID.ONSITE_GROUPS;
+        return ((onsiteGroup && isLeader) || (!onsiteGroup));
+    }
 
-   //From groupDetail.participants.controller.js loadGroupParticipants()
```
Comment (Doug-Shannon) [inline crossroads.net/app/group_tool/services/groupDetail.service.js]: stupid github understanding of what has changed, the comment is still in the file on line 9

Comment (JoeKer1) [inline crossroads.net/app/group_tool/services/groupDetail.service.js]: Ahh I see it looked like code for a sec

File: Gateway/crds-angular/Controllers/API/GroupController.cs (hunk 1 above)
Comment (JoeKer1) [inline Gateway/crds-angular/Controllers/API/GroupController.cs]: As part of this PR?

File: crossroads.net/spec-es6/group_tool/services/groupDetail.service.spec.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    
+    it('should return true, user is not leader but it is not Onsite Group', () => {
+        let isLeader = false;
+        let groupTypeId = 7732323;
+
+        let result = fixture.canAccessParticipants(isLeader, groupTypeId);
+
+        expect(result).toBeTruthy();
+    });
+
```
Comment (JoeKer1) [inline crossroads.net/spec-es6/group_tool/services/groupDetail.service.spec.js]: I love these tests all the bests tests.

**Pull request merged**